# Inference

Notebook ini digunakan untuk melakukan inference atau prediksi pada data teks baru menggunakan model LSTM berbasis BoW yang telah dilatih sebelumnya. Proses meliputi load model dan vectorizer, transformasi teks, serta menghasilkan label prediksi beserta confidence score.

## Import Library

In [2]:
from google.colab import files
import joblib
import numpy as np
from tensorflow.keras.models import load_model

## Load Model dan Vectorizer

In [3]:
model = load_model('model_lstm_bow.h5')
bow = joblib.load('bow_vectorizer.pkl')

In [4]:
mapping = {0: 'Negative', 1: 'Neutral', 2: 'Positive'}

## Definisi Fungsi Prediksi

In [5]:
def prediksi(teks):
    X = bow.transform([teks]).toarray()
    X = X.reshape(1, 1, -1)
    prob = model.predict(X, verbose=0)[0]
    kelas = np.argmax(prob)

    # Tampilkan hasil
    label = mapping[kelas]
    confidence = prob[kelas]

    return label, confidence, prob

## Data Uji (Test Cases)

In [6]:
test_cases = [
    "saya tau ini kesalahan saya karna nomor saya mati saya juga sudah minta bantuan ke dana dan saya juga sudah mengikuti semua prosedur yg ada tapi kenapa saldo dana saya tetap tidak di kembalikan ke nomor yg baru"
]

## Hasil Prediksi

In [7]:
print("HASIL PREDIKSI:")
for teks in test_cases:
    label, conf, prob = prediksi(teks)
    print(f"\nTeks: {teks}")
    print(f"Prediksi: {label} (confidence: {conf:.4f})")

HASIL PREDIKSI:

Teks: saya tau ini kesalahan saya karna nomor saya mati saya juga sudah minta bantuan ke dana dan saya juga sudah mengikuti semua prosedur yg ada tapi kenapa saldo dana saya tetap tidak di kembalikan ke nomor yg baru
Prediksi: Negative (confidence: 0.8077)


Model berhasil mengklasifikasikan teks dengan baik. Teks diprediksi sebagai Negative dengan confidence sebesar 0.8077, yang menunjukkan adanya keluhan atau ketidakpuasan terkait layanan.